starknet_py/utils/typed_data_test.py

```diff
@@ -197,15 +197,75 @@ def _make_typed_data(included_type: str, revision: Revision):
 
 
 @pytest.mark.parametrize(
-    "included_type",
+    "included_type, revision",
     [
-        "felt",
-        "felt*",
-        "string",
-        "selector",
-        "merkletree"
+        ("", Revision.V1),
+        ("myType*", Revision.V1)
     ],
 )
-def test_invalid_types(included_type: str):
+def test_invalid_type_names(included_type: str, revision: Revision):
+    with pytest.raises(ValueError):
+        _make_typed_data(included_type, revision)
+
+
+@pytest.mark.parametrize(
+    "included_type, revision",
+    [
+        ("felt", Revision.V0),
+        ("string", Revision.V0),
+        ("selector", Revision.V0),
+        ("merkletree", Revision.V0),
+    ],
+)
+def test_types_redefinition(included_type: str, revision: Revision):
     with pytest.raises(ValueError, match=f"Reserved type name: {included_type}"):
-        _make_typed_data(included_type, Revision.V1)
+        _make_typed_data(included_type, revision)
+
+
+def test_custom_type_definition():
+    _make_typed_data("myType", Revision.V0)
```

Comment:
What's the purpose of this test?

Reply:
It tests whether it's possible to create type data with custom type as primary one. In Swift SDK there is a [similar test](https://github.com/software-mansion/starknet.swift/pull/170/files#diff-c38c472f61f02b67b364e318b6fa8fed796f64c1bc5bf21a2b065b9d55bcf442R92).